crates/rpc-types/src/net.rs

```diff
@@ -100,3 +100,54 @@ pub struct PeerStats {
     pub banned: u32,
     pub known: u32,
 }
+
+/// Represents the connectivity state of a peer in a network, indicating the reachability and
+/// interaction status of a node with its peers.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Copy)]
+#[serde(rename_all = "camelCase")]
```

Comment:
We'll want to add the u8 representation. Also should probably mark `NotConnected` as the `#[default]`

```rust
#[repr(u8)]
```